compiler/rustc_interface/src/passes.rs

```diff
@@ -565,6 +565,11 @@ fn resolver_for_lowering<'tcx>(
 }
 
 fn write_dep_info(tcx: TyCtxt<'_>, (): ()) {
+    // Make sure name resolution and macro expansion is run for
+    // the side-effect of providing a complete set of all
+    // accessed files and env vars.
+    let _ = tcx.resolver_for_lowering(());
```

Comment:
This should use `ensure()`.

Reply:
If  `resolver_for_lowering` weren't `eval_always` and it is green, using `ensure()` would mean that the side effects of `resolver_for_lowering` (adding the list of used files and env vars to the `Session`) are not replayed as far as I understand it. Now `resolver_for_lowering` is `eval_always` currently, so it shouldn't matter for now I guess. There shouldn't be any perf difference though as `resolver_for_lowering` returns a reference.

Reply:
Should we maybe add a comment in the query definition explaining this, i.e. why this needs to be `eval_always`?